plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -202,11 +203,12 @@ public static LongWriteFunction shortDecimalWriteFunction(DecimalType decimalTyp
     {
         requireNonNull(decimalType, "decimalType is null");
         checkArgument(decimalType.isShort());
-        return (statement, index, value) -> {
+
+        return LongWriteFunction.of(Types.DECIMAL, (statement, index, value) -> {
```

Comment:
How is Types.DECIMAL translated into a type on the remote side?
For example Trino would infer `decimal(38,0)`. 
Now, if there is a sibling value with type eg `decimal(8,4)` they may get coerced together to a common type (`decimal(38,0)`), which could result in trailing digits being lost.

Also, does it have any interaction with statement batching on the remote server?
Such a rewrite from single-row INSERT to multi-row INSERT would make such coercions easier.
Even without batching this could occur e.g. for ARRAYs, but maybe not only.

Reply:
Since this is used only for NULLs it doesn't matter. In Oracle JDBC driver, this type is only the information which null binder to use - every numeric type uses the same null binder.

Case values are `java.sql.Types`: 

<img width="231" alt="Screenshot 2022-05-19 at 21 39 22" src="https://user-images.githubusercontent.com/66972/169388952-0a57fee3-e68a-462a-a630-afbd19812e2f.png">

They all match to a single `VarnumNullBinder` (which doesn't depend on actual `java.sql.Types` type).

So `BIT`, `BOOLEAN` (internally mapped to `BIT` as well), `TINYINT`, `BIGINT`, `NUMERIC`, `DECIMAL`, `INTEGER`, `SMALLINT`, `FLOAT`, `REAL`, `DOUBLE` have exactly the same `NULL` representation in Oracle, being sent over the database protocol.

Reply:
> Since this is used only for NULLs it doesn't matter.

I understand how this is used. My point was -- a type inferred for a NULL value may affect how a sibling non-NULL value gets processed, when a coercion to a common super type is applied.